go/go-sql.go

```diff
@@ -23,37 +23,52 @@ import (
 	"reflect"
 	"runtime"
 	"strings"
+
+	"go.opentelemetry.io/otel/propagation"
 )
 
 const (
-  
-  
-  
-  string = "route"
-  controller string = "controller"
-  action     string = "action"
-  framework  string = "framework"
-  driver     string = "driver"
+	route       string = "route"
+	controller  string = "controller"
+	action      string = "action"
+	framework   string = "framework"
+	driver      string = "driver"
+	traceparent string = "traceparent"
 )
 
 type DB struct {
 	*sql.DB
-	options CommenterOptions
+	options         CommenterOptions
+	otelTraceparent string
 }
 
 type CommenterOptions struct {
-	EnableDBDriver   bool
-	EnableRoute      bool
-	EnableFramework  bool
-	EnableController bool
-	EnableAction     bool
+	EnableDBDriver    bool
+	EnableRoute       bool
+	EnableFramework   bool
+	EnableController  bool
+	EnableAction      bool
+	EnableTraceparent bool
 }
 
 func Open(driverName string, dataSourceName string, options CommenterOptions) (*DB, error) {
 	db, err := sql.Open(driverName, dataSourceName)
 	return &DB{DB: db, options: options}, err
 }
 
+func (db *DB) AddorUpdateCurrentSpan(ctx context.Context) {
```

Comment:
I assume this context is from the API request call and hence should be passed to QueryContext/ExecContext method. 
So they would anyway contain the context values, we can check if that value is present and pull that from that, add it to SQL Comment.